In [2]:
import pandas as pd
import matplotlib.pyplot as plt

In [3]:
df = pd.read_excel("metaClean43Brightspace.xlsx")

df.head()

,url,title,studio,rating,runtime,cast,director,genre,summary,awards,metascore,userscore,RelDate
0,https://www.metacritic.com/movie/!women-art-re...,!Women Art Revolution,Hotwire Productions,| Not Rated,83.0,NaN,Lynn Hershman-Leeson,Documentary,NaN,NaN,70,NaN,2011-06-01
1,https://www.metacritic.com/movie/10-cloverfiel...,10 Cloverfield Lane,Paramount Pictures,| PG-13,104.0,"John Gallagher Jr.,John Goodman,Mary Elizabeth...",Dan Trachtenberg,"Action,Sci-Fi,Drama,Mystery,Thriller,Horror","Waking up from a car accident, a young woman (...","#18MostDiscussedMovieof2016 , #1MostSharedMovi...",76,7.7,2016-03-11
2,https://www.metacritic.com/movie/10-items-or-less,10 Items or Less,Click Star,| R,82.0,"Jonah Hill,Morgan Freeman,Paz Vega",Brad Silberling,"Drama,Comedy,Romance",While researching a role as a supermarket mana...,NaN,54,5.8,2006-12-01
3,https://www.metacritic.com/movie/10-years,10 Years,Anchor Bay Entertainment,| R,100.0,"Channing Tatum,Chris Pratt,Jenna Dewan",Jamie Linden,"Drama,Comedy,Romance",NaN,NaN,61,6.9,2012-09-14
4,https://www.metacritic.com/movie/100-bloody-acres,100 Bloody Acres,Music Box Films,| Not Rated,91.0,NaN,Cameron Cairnes,"Horror,Comedy",Reg and Lindsay run an organic fertilizer busi...,NaN,63,7.5,2013-06-28


Here I read the excel file, I firstly imported openpyxl to my environment so I was able to read an excel file. Then I read the excelfile and printed out the first 5 rows.

In [4]:
df.info()
df.isnull().sum()

<class 'pandas.DataFrame'>
RangeIndex: 11364 entries, 0 to 11363
Data columns (total 13 columns):
 #   Column     Non-Null Count  Dtype         
---  ------     --------------  -----         
 0   url        11364 non-null  str           
 1   title      11364 non-null  str           
 2   studio     11014 non-null  str           
 3   rating     10297 non-null  str           
 4   runtime    11109 non-null  float64       
 5   cast       7662 non-null   str           
 6   director   11350 non-null  str           
 7   genre      11344 non-null  str           
 8   summary    5467 non-null   str           
 9   awards     4387 non-null   str           
 10  metascore  11364 non-null  int64         
 11  userscore  9259 non-null   float64       
 12  RelDate    11364 non-null  datetime64[us]
dtypes: datetime64[us](1), float64(2), int64(1), str(9)
memory usage: 1.1 MB


url             0
title           0
studio        350
rating       1067
runtime       255
cast         3702
director       14
genre          20
summary      5897
awards       6977
metascore       0
userscore    2105
RelDate         0
dtype: int64

Here I checked the info of the datasets, to see the rows and how many non-null values they have, I also checked the datatypes. I did a double check on how many null values there are in each columm. You can see that url and title got 0 null values, but summary and awards got more than half of the rows for null values. 

In [5]:
df['rating'].unique

<bound method Series.unique of 0        | Not Rated
1            | PG-13
2                | R
3                | R
4        | Not Rated
            ...     
11359        | PG-13
11360           | PG
11361    | Not Rated
11362           | PG
11363            NaN
Name: rating, Length: 11364, dtype: str>

I saw that the rating columm had a problem that some of them started with '|', now we need to remove that.

In [6]:
df['rating'] = df['rating'].str.strip(' | ').str.strip()
df['rating'].unique

<bound method Series.unique of 0        Not Rated
1            PG-13
2                R
3                R
4        Not Rated
           ...    
11359        PG-13
11360           PG
11361    Not Rated
11362           PG
11363          NaN
Name: rating, Length: 11364, dtype: str>

In [7]:
df['awards'] = df['awards'].fillna('none')
df['awards'].unique

<bound method Series.unique of 0                                                     none
1        #18MostDiscussedMovieof2016 , #1MostSharedMovi...
2                                                     none
3                                                     none
4                                                     none
                               ...                        
11359    #87MostDiscussedMovieof2016 , #80MostSharedMov...
11360                                                 none
11361                                                 none
11362    #80BestMovieof2016 , #11MostDiscussedMovieof20...
11363                                                 none
Name: awards, Length: 11364, dtype: str>

For the nullvalues in the colomm awards we can assume that the movie did not win any awards. Therefore I changed the null value to none.

In [8]:
df['studio'] = df['studio'].str.strip()
df['studio'].value_counts()

studio
IFC Films                                  408
Universal Pictures                         345
Sony Pictures Classics                     344
Warner Bros. Pictures                      302
Netflix                                    301
                                          ... 
Manga Video                                  1
Real Women Make Waves                        1
Film Desk, The                               1
myCinema                                     1
20th Century Fox International Classics      1
Name: count, Length: 1118, dtype: int64

I still need to clean the runtime and userscore, I want to use median is that okay? Than there are some rows where director and genre is missing it is not a lot, is it possible to drop them or can we leave them open? There are studios who are written in more ways, what is the best way to clean this? 

In [9]:
df['runtime'] = df['runtime'].fillna(df['runtime'].median())
df['userscore'] = df['userscore'].fillna(df['userscore'].median())

Here I filled the nullvalues of runtime and metascore with the median.

In [10]:
df[df['genre'].isnull()]

,url,title,studio,rating,runtime,cast,director,genre,summary,awards,metascore,userscore,RelDate
1798,https://www.metacritic.com/movie/chalk-2000,Chalk,Tenderloin Action Group,NaN,134.0,NaN,Rob Nilsson,NaN,This gritty drama focuses on the misfits who h...,none,57,6.7,2000-08-25
1886,https://www.metacritic.com/movie/chop-shop,Chop Shop,Koch Lorber Films,Unrated,84.0,NaN,Ramin Bahrani,NaN,NaN,#18BestMovieof2008,83,7.7,2008-02-27
1968,https://www.metacritic.com/movie/closed-curtain,Closed Curtain,Variance Films,Not Rated,106.0,NaN,Jafar Panahi,NaN,NaN,#50BestMovieof2014,81,5.1,2014-07-09
2300,https://www.metacritic.com/movie/daydream-beli...,Daydream Believer,Film Threat DVD,NaN,79.0,NaN,Debra Eisenstadt,NaN,"An edgy, contemporary take on an age-old story...",none,53,6.7,2002-09-25
2502,https://www.metacritic.com/movie/djomeh,Djomeh,New Yorker Films,NaN,94.0,"Jalil Nazari,Rashid Akbari",Hassan Yektapanah,NaN,"Djomeh, a shy young Afghani working in a remot...",#52BestMovieof2001,76,6.7,2001-09-05
2969,https://www.metacritic.com/movie/farewell-amor,Farewell Amor,IFC Films,NR,101.0,NaN,NaN,NaN,NaN,none,75,6.7,2020-12-11
4215,https://www.metacritic.com/movie/in-heaven-und...,In Heaven Underground: The Weissensee Jewish C...,Seventh Art Releasing,Not Rated,90.0,NaN,NaN,NaN,NaN,none,67,6.7,2011-11-18
4844,https://www.metacritic.com/movie/langrishe-go-...,Langrishe Go Down,Castle Hill Productions,NaN,105.0,NaN,David Hugh Jones,NaN,The theatrical release of a 1978 BBC film star...,none,59,4.0,2002-07-17
4957,https://www.metacritic.com/movie/les-miserable...,Les Misérables,Amazon Studios,Not Rated,102.0,"Alexis Manenti,Damien Bonnard,Djibril Zonga,Je...",Ladj Ly,NaN,NaN,#80BestMovieof2020,78,6.6,2020-01-10
4986,https://www.metacritic.com/movie/liberte,Liberté,Cinema Guild,Not Rated,132.0,"Baptiste Pinteaux,Catalin Jugravu,Francesc Dar...",Albert Serra,NaN,NaN,none,51,6.7,2020-05-01


Here we took a look at all the rows who had a nullvalue in genre, we did this to see if the movies are important to keep.

In [11]:
df[df['director'].isnull()]

,url,title,studio,rating,runtime,cast,director,genre,summary,awards,metascore,userscore,RelDate
1578,https://www.metacritic.com/movie/brothers-in-arms,Brothers in Arms,First Run Features,Unrated,68.0,NaN,NaN,War,This documentary focuses on the crew of Patrol...,none,48,6.7,2004-08-27
2969,https://www.metacritic.com/movie/farewell-amor,Farewell Amor,IFC Films,NR,101.0,NaN,NaN,NaN,NaN,none,75,6.7,2020-12-11
3364,https://www.metacritic.com/movie/generation-war,Generation War,Music Box Films,Not Rated,270.0,NaN,NaN,"Drama,History,War",NaN,none,57,7.4,2014-01-15
3759,https://www.metacritic.com/movie/head-games,Head Games,Variance Films,PG--13,100.0,NaN,NaN,Documentary,NaN,none,69,6.7,2012-09-21
4215,https://www.metacritic.com/movie/in-heaven-und...,In Heaven Underground: The Weissensee Jewish C...,Seventh Art Releasing,Not Rated,90.0,NaN,NaN,NaN,NaN,none,67,6.7,2011-11-18
4331,https://www.metacritic.com/movie/into-eternity,Into Eternity,International Film Circuit,Unrated,75.0,NaN,NaN,Documentary,NaN,none,68,6.7,2011-02-02
5253,https://www.metacritic.com/movie/lynch,Lynch,Absurda,Not Rated,84.0,NaN,NaN,Documentary,NaN,none,67,8.6,2007-10-26
6668,https://www.metacritic.com/movie/price-check,Price Check,IFC Films,Not Rated,92.0,NaN,NaN,"Drama,Comedy",A mid-level exec in a supermarket chain (Mabiu...,none,67,6.5,2012-11-16
6882,https://www.metacritic.com/movie/red-hill,Red Hill,Strand Releasing,R,95.0,"Ryan Kwanten,Steve Bisley,Tommy Lewis",NaN,"Thriller,Crime,Western",NaN,none,62,6.7,2010-11-05
7311,https://www.metacritic.com/movie/seven-days-in...,Seven Days in Utopia,Utopia Pictures,G,120.0,NaN,NaN,"Drama,Sport",NaN,none,37,5.8,2011-09-02


Now we did the same for directors. 

In [12]:
df = df.dropna(subset=['director', 'genre'])

Here I dropped the rows that have nullvalues for director and genre.

In [13]:
df['rating'] = df['rating'].replace({'PG--13': 'PG-13', 'PG-13`': 'PG-13'})

There were still some values of rating with a double -- and one with some with a '`' I changed these one so everyone of them were the same value.

In [14]:
df['title_clean'] = df['title'].str.strip().str.lower()
df.head()

,url,title,studio,rating,runtime,cast,director,genre,summary,awards,metascore,userscore,RelDate,title_clean
0,https://www.metacritic.com/movie/!women-art-re...,!Women Art Revolution,Hotwire Productions,Not Rated,83.0,NaN,Lynn Hershman-Leeson,Documentary,NaN,none,70,6.7,2011-06-01,!women art revolution
1,https://www.metacritic.com/movie/10-cloverfiel...,10 Cloverfield Lane,Paramount Pictures,PG-13,104.0,"John Gallagher Jr.,John Goodman,Mary Elizabeth...",Dan Trachtenberg,"Action,Sci-Fi,Drama,Mystery,Thriller,Horror","Waking up from a car accident, a young woman (...","#18MostDiscussedMovieof2016 , #1MostSharedMovi...",76,7.7,2016-03-11,10 cloverfield lane
2,https://www.metacritic.com/movie/10-items-or-less,10 Items or Less,Click Star,R,82.0,"Jonah Hill,Morgan Freeman,Paz Vega",Brad Silberling,"Drama,Comedy,Romance",While researching a role as a supermarket mana...,none,54,5.8,2006-12-01,10 items or less
3,https://www.metacritic.com/movie/10-years,10 Years,Anchor Bay Entertainment,R,100.0,"Channing Tatum,Chris Pratt,Jenna Dewan",Jamie Linden,"Drama,Comedy,Romance",NaN,none,61,6.9,2012-09-14,10 years
4,https://www.metacritic.com/movie/100-bloody-acres,100 Bloody Acres,Music Box Films,Not Rated,91.0,NaN,Cameron Cairnes,"Horror,Comedy",Reg and Lindsay run an organic fertilizer busi...,none,63,7.5,2013-06-28,100 bloody acres


Here I normalised the title columm, i made sure that there were only lower cases in the titles and added that to the new colomm title_clean

In [15]:
df['match_key'] = df['title_clean'] + '_' + df['RelDate'].dt.year.astype(str)
df['match_key'].head()

0    !women art revolution_2011
1      10 cloverfield lane_2016
2         10 items or less_2006
3                 10 years_2012
4         100 bloody acres_2013
Name: match_key, dtype: str

Here I made a match_key so that I eventually can make the new id where I can math on. This because I don't want any letters in my id (primary keys)

In [16]:
df['match_key'].duplicated().sum()

np.int64(3)

I was checking for duplicates in the new colomm and it gave me three, so now im gonna check which ones they are.

In [17]:
double_keys = df[df['match_key'].duplicated(keep=False)]
double_keys.sort_values('match_key')

,url,title,studio,rating,runtime,cast,director,genre,summary,awards,metascore,userscore,RelDate,title_clean,match_key
2625,https://www.metacritic.com/movie/dreamland-2019,Dreamland,Paramount Pictures,R,98.0,"Darby Camp,Finn Cole,Garrett Hedlund,Ginger T....",Miles Joris-Peyrafitte,"Drama,Thriller",NaN,none,57,5.6,2020-11-13,dreamland,dreamland_2020
2626,https://www.metacritic.com/movie/dreamland-2020,Dreamland,Uncork'd Entertainment,TV-MA,92.0,"Astrid Roos,Barbara Hellemans,Daphné Huynh,Gu...",Bruce McDonald,"Fantasy,Horror,Comedy,Crime",On the night of the strangest weddings in hist...,none,50,6.7,2020-06-05,dreamland,dreamland_2020
7975,https://www.metacritic.com/movie/swan-song,Swan Song,Magnolia Pictures,Not Rated,105.0,"Annie Kitral,Dave Sorboro,Eric Eisenbrey,Ira H...",Todd Stephens,Drama,NaN,none,66,6.8,2021-08-06,swan song,swan song_2021
7976,https://www.metacritic.com/movie/swan-song-2021,Swan Song,Apple TV+,R,116.0,"Adam Beach,Awkwafina,Dax Rey,Glenn Close,Jayr ...",Benjamin Cleary,"Sci-Fi,Drama","In the near future, Cameron Turner (Mahershala...",none,66,7.0,2021-12-17,swan song,swan song_2021
9058,https://www.metacritic.com/movie/the-hunter,The Hunter,Olive Films,Not Rated,90.0,NaN,Rafi Pitts,Thriller,NaN,#66BestMovieof2012,76,7.1,2012-01-04,the hunter,the hunter_2012
9059,https://www.metacritic.com/movie/the-hunter-2012,The Hunter,Magnolia Pictures,R,100.0,"Jacek Koman,Morgana Davies,Sam Neill,Willem Dafoe",Daniel Nettheim,"Adventure,Drama",NaN,none,63,7.1,2012-04-06,the hunter,the hunter_2012


Here I did the extra check to see if the duplicates really are the same movies. As you can see it definently are different movies, with different studios,casts and directors etc. So I'm gonna leave this duplicates, but I need to add something to them otherwise they will all get matched to the other id's. Therefore I'm gonna add the runtime, this because in the sales data runtime comes up to, this colomm is the most unique and all the runtimes differ from eachother so that is the best one to match on. 

In [18]:
colliding_keys = df['match_key'][df['match_key'].duplicated(keep=False)]

df.loc[df['match_key'].isin(colliding_keys), 'match_key'] = (
    df.loc[df['match_key'].isin(colliding_keys), 'match_key'] 
+ '_' + df.loc[df['match_key'].isin(colliding_keys), 'runtime'].astype(str)
)    

In [19]:
df['match_key'].duplicated().sum()

np.int64(0)

So now i added the runtime to the duplicate values and there are 0 nullvalues left. 

In [20]:
df.to_csv('metaclean_cleaned.csv', index=False) 

In [21]:
df.insert(0, "movie_id", range(1, len(df) + 1))

Here I made a movie_id for the primary key to connect all the tables with eachother 

In [22]:
print(df[["movie_id", "title", "match_key"]].head(5))

   movie_id                  title                   match_key
0         1  !Women Art Revolution  !women art revolution_2011
1         2    10 Cloverfield Lane    10 cloverfield lane_2016
2         3       10 Items or Less       10 items or less_2006
3         4               10 Years               10 years_2012
4         5       100 Bloody Acres       100 bloody acres_2013


In [23]:
df.to_csv('metaclean_cleaned.csv', index=False)


In [24]:
df = pd.read_csv('metaclean_cleaned.csv')

# Bij de 2 films met meerdere regisseurs pakken we de eerste
df["director_first"] = df["director"].str.split(",").str[0].str.strip()

# DIRECTOR-tabel: unieke regisseurs met een eigen ID
unique_directors = sorted(df["director_first"].unique())
director = pd.DataFrame({
    "director_id": range(1, len(unique_directors) + 1),
    "director_name": unique_directors,
})

# director_id als FK in MOVIE; de naam hoort alleen in DIRECTOR
df = df.merge(director, left_on="director_first", right_on="director_name", how="left")
df = df.drop(columns=["director_first", "director_name"])

director.to_csv("director.csv", index=False)
df.to_csv("metaclean_v2.csv", index=False)

print(director.shape)
print(df[["movie_id", "director_id"]].head(3))
print(df.director_id.isna().sum())   # moet 0 zijn

(6126, 2)
   movie_id  director_id
0         1         3790
1         2         1379
2         3          803
0


Here I created the DIRECTOR table and linked it to MOVIE. In the ERD there is a 1:N relation between DIRECTOR and MOVIE, so one director can have multiple movies, and that means the foreign key has to go on the movie side. So I don't need a separate junction table for this one, I just add a director_id column to the movie table, which is an FK that points to DIRECTOR. The movie_id stays the PK of MOVIE. First I checked how many movies have more than one director, and it turns out this is only the case for 2 out of the 11,332 movies. Because it's such a small amount, I decided to just take the first director that is mentioned for those movies, so we can stay within the 1:N structure of the ERD. This is a small deviation from the raw data, so I'm gonna mention it in the project document as well.

After that I took all the unique director names and gave each of them their own director_id (1, 2, 3 etc.), which gives me the DIRECTOR table with director_id and director_name. This resulted in 6,126 unique directors. Then I merged the director_id back into the main dataframe on the director name and dropped the helper columns afterwards, so the name only lives in the DIRECTOR table and not in MOVIE as well. I also checked that every one of the 11,332 movies got a director_id and that none of them is missing. In the end I saved two files: director.csv for the DIRECTOR table and metaclean_v2.csv, which is the movie data with the extra director_id column.

In [25]:
df = pd.read_csv('metaclean_v2.csv')   

exploded = (
    df[["movie_id", "genre"]]
    .assign(genre=df["genre"].str.split(","))
    .explode("genre")
)
exploded["genre"] = exploded["genre"].str.strip()
exploded = exploded.dropna(subset=["genre"])
exploded = exploded[exploded["genre"] != ""]

print(len(exploded), "rows before removing duplicates")
print(exploded.duplicated().sum(), "duplicate rows")

dupes = exploded[exploded.duplicated(keep=False)].drop_duplicates()
print(dupes.merge(df[["movie_id", "title", "genre"]], on="movie_id",
                  suffixes=("_split", "_original")))

exploded = exploded.drop_duplicates()

print(len(exploded), "rows after removing duplicates")  

unique_genres = sorted(exploded["genre"].unique())
genre = pd.DataFrame({
    "genre_id": range(1, len(unique_genres) + 1),
    "genre": unique_genres
})

movie_genre = exploded.merge(genre, on="genre")[["movie_id", "genre_id"]]

genre.to_csv("genre.csv", index=False)
movie_genre.to_csv("movie_genre.csv", index=False)

print(genre.shape)
print(movie_genre.shape)
print(movie_genre.duplicated().sum())


27033 rows before removing duplicates
4 duplicate rows
   movie_id genre_split               title                   genre_original
0       718       Adult     Anatomy of Hell                Drama,Adult,Adult
1      3866       Adult     History Lessons  History,Documentary,Adult,Adult
2      4319       Adult            Intimacy                Drama,Adult,Adult
3      8829       Adult  The Girl Next Door          Documentary,Adult,Adult
27029 rows after removing duplicates
(27, 2)
(27029, 2)
0


Here I created the GENRE table and the MOVIE_GENRE junction table. In the ERD these are two separate tables, because a movie can have more than one genre and a genre belongs to a lot of movies. That is a many-to-many relation, so I can't just put a genre_id column in the movie table like I did for director. I first checked the genre column and there are no missing values. It turns out that 7,871 movies have more than one genre, and they are all in one text cell separated by commas. So I split the genre column on the comma and gave every genre of a movie its own row.

While doing this I noticed that for 4 movies (for example Anatomy of Hell and Intimacy) the genre Adult is listed twice in the same cell. If I leave this as it is, the junction table gets duplicate rows and that would give an error in PostgreSQL as soon as I set movie_id and genre_id together as the primary key. Therefore I removed these duplicates before building the tables.

After that I took all the unique genres and gave each of them their own genre_id (1, 2, 3 etc.), which gives me the GENRE table with genre_id and genre. This resulted in 27 unique genres. Then I merged this back with the split data on the genre name, so I could link every movie_id to the right genre_id. This is the MOVIE_GENRE junction table, and it has 27,029 rows. I also checked that there are no duplicate rows left in it and that all 11,332 movies are in there. In the end I saved two files: genre.csv for the GENRE table and movie_genre.csv for the junction table.

In [26]:
df = pd.read_csv('metaclean_v2.csv')
sales = pd.read_csv('sales_cleaned.csv')

print(sales["match_key"].isna().sum(), "sales rows without match_key")
print(sales["match_key"].dropna().duplicated().sum(), "duplicate match_keys with a title")

movie_sales = (
    df[["movie_id", "match_key"]]
    .merge(sales[["sales_id", "match_key"]], on="match_key")
    [["movie_id", "sales_id"]]
)

print(movie_sales.shape)
print(movie_sales["movie_id"].is_unique)
print(movie_sales["sales_id"].is_unique)
print(movie_sales.duplicated().sum())
print((~sales["match_key"].isin(df["match_key"])).sum(), "sales rows without a match in meta")

movie_sales.to_csv("movie_sales.csv", index=False)

8 sales rows without match_key
0 duplicate match_keys with a title
(7978, 2)
True
True
0
22634 sales rows without a match in meta


Here I created the MOVIE_SALES junction table that connects MOVIE to SALES. The two datasets don't share an ID, so I used the match_key (the cleaned title combined with the release year) that I created earlier in both files. Before joining I checked the sales data and found 8 rows without a title, and therefore without a match_key. These can't be matched to any movie, so they simply drop out of the join. I also checked that no match_key with a title appears more than once in the sales data, so a movie can't accidentally get linked to multiple sales rows.

After that I merged movie_id and match_key from the movie data with sales_id and match_key from the sales data, and kept only movie_id and sales_id. This resulted in 7,978 rows. I checked that there are no duplicate rows in it and that both movie_id and sales_id only appear once, so for now it works as a 1:1 link, but it stays a junction table because that is how it is drawn in the ERD. One thing to keep in mind is that only 7,978 of the 30,612 sales rows could be linked to a movie. The other 22,634 sales rows stay in the SALES table but have no row in MOVIE_SALES, which means the analysis can only use the movies that are linked. In the end I saved the result as movie_sales.csv.

In [28]:
df = pd.read_csv('metaclean_v2.csv')
sales = pd.read_csv('sales_cleaned.csv')

runtime_sales = (
    df[["movie_id", "match_key"]]
    .merge(sales[["match_key", "runtime"]], on="match_key")
    [["movie_id", "runtime"]]
    .rename(columns={"runtime": "runtime_sales"})
)

movie =df.merge(runtime_sales, on="movie_id", how="left")
movie = movie.rename(columns={"runtime": "runtime_meta"})

movie = movie[["movie_id", "director_id", "url", "title", "studio", "rating", "runtime_meta", "runtime_sales", "metascore", "userscore", "RelDate"]]

movie["runtime_meta"] = movie["runtime_meta"].astype("Int64")
movie["runtime_sales"] = movie["runtime_sales"].round().astype("Int64")

movie.insert(2, "actor_id", pd.array([pd.NA] * len(movie), dtype="Int64"))

movie.to_csv("movie.csv", index=False)

print(movie.shape)
print(movie["movie_id"].is_unique)
print(movie["runtime_sales"].notna().sum(), "movies with a runtime from sales")
print(movie["actor_id"].isna().all())
print(movie.columns.tolist()) 

(11332, 12)
True
6367 movies with a runtime from sales
True
['movie_id', 'director_id', 'actor_id', 'url', 'title', 'studio', 'rating', 'runtime_meta', 'runtime_sales', 'metascore', 'userscore', 'RelDate']


Here I created the MOVIE table. This is basically the metaclean data, but then reduced to only the columns that are in the ERD. The columns cast, director, genre, summary, awards, title_clean and match_key are dropped, because cast, director and genre now live in their own tables and the rest are only helper columns or text that we don't need in the database. I also renamed runtime to runtime_meta, because the ERD has two runtime columns, one from Metacritic and one from the sales data.

For runtime_sales I used the match_key again to link every movie to its row in the sales data and took the runtime from there. This resulted in 6,367 movies with a runtime_sales. The other movies get an empty value, because either the movie has no match in the sales data or the sales data has no runtime for it. I don't think this is a problem, but it is good to keep in mind if we use this column in the analysis. Both runtimes are also converted to whole numbers, because the ERD defines them as integers.

The ERD also has an actor_id column in MOVIE. The ACTOR table is made by my teammate, so I can't fill this column yet. For now I added it as an empty column in the right position, and as soon as the ACTOR table is finished I can fill it. Finally I checked that the table has 11,332 rows, that every movie_id is unique, and that the columns are in the same order as in the ERD. I saved the result as movie.csv.